cremalab/workout-tracker
# Create account sign up page

## Reference
- [Original Asana Task](https://app.asana.com/0/622910146909793/622910146909803) 
- [Related Task](https://app.asana.com/0/622910146909793/622910146909802)

## Overview

Builds on the Hapi/MongoDB base that's currently on the develop branch. User entered sign up data now saves to mongo.

## To-Do
Future enhancements needed:
- [ ] Add validation for email and password fields with Joi
- [x] Perform password validation on client side without global variables
- [ ] Add password encryption with bcrypt
- [ ] Use UI library 

## Testing Instructions

- npm install  && cd client && npm install && cd ..
- npm start
- localhost:3000 should open up in your browser and API server localhost:5000 should connect as well
- Entering in an invalid email should trigger native email check on Submit
- Entering mismatched passwords should trigger error message on Submit
- Entering valid data should save data to mongo
